subsys/bluetooth/mesh/adv.c

```diff
@@ -41,8 +41,15 @@ const uint8_t bt_mesh_adv_type[BT_MESH_ADV_TYPES] = {
 
 K_FIFO_DEFINE(bt_mesh_adv_queue);
 
+static void adv_buf_destroy(struct net_buf *buf)
+{
+	bt_mesh_adv_send_end(0, BT_MESH_ADV(buf));
```

Comment:
Great thinking.

Reply:
Try to avoid call end if not started.
``` C
    if (adv.started) {
        bt_mesh_adv_send_end(0, BT_MESH_ADV(buf));
    }
```
And try avoid call end if failed to started:

``` C

    // in bt_mesh_adv_send_start
    if (err) {
        adv->cb = NULL;
    }
```